pages/common/xargs.md

```diff
@@ -1,19 +1,20 @@
 # xargs
 
-> Execute a command with piped arguments.
+> Execute a command with piped arguments coming from another command, a file, etc.
+> The input is treated as a single block of text and split into separate arguments on spaces, tabs, newlines and end-of-file.
 
-- Main use:
+- Main usage pattern:
 
-`{{arguments}} | xargs {{command}}`
+`{{arguments_source}} | xargs {{command}}`
 
-- Specific example: delete all files that start with 'M':
+- Delete all files with a `.backup` extension:
 
-`find . -name 'M*' | xargs rm`
+`{{find . -name '*.backup'}} | xargs {{rm -v}}`
 
-- Handle whitespace in arguments:
+- Use the NUL (`\0`) character as separator (useful to split the input on newlines but not on spaces):
```

Comment:
>  I understand the value on specifying that the input is split on newlines

We can say `Use the NUL (`\0`) character as separator to split input lines (useful if the input to xargs contains spaces):`
> since xargs' -0 was precisely made to work with find's -print0

That is true actually. Well, there is no harm in giving `find ... -print0` itself, we just have to be a bit more verbose and explain what `print0` does. And practically speaking, `xargs -0` is most used along with `find ... -print0`. If not, any other command is also fine by me.